extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/auth/HashicorpVaultAuthenticationExtension.java

```diff
@@ -14,23 +14,22 @@
 
 package org.eclipse.edc.vault.hashicorp.auth;
 
-import org.eclipse.edc.runtime.metamodel.annotation.Configuration;
 import org.eclipse.edc.runtime.metamodel.annotation.Extension;
 import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
 import org.eclipse.edc.spi.system.ServiceExtension;
-import org.eclipse.edc.vault.hashicorp.client.HashicorpVaultSettings;
 import org.eclipse.edc.vault.hashicorp.spi.auth.HashicorpVaultTokenProvider;
 
 @Extension(value = HashicorpVaultAuthenticationExtension.NAME)
 public class HashicorpVaultAuthenticationExtension implements ServiceExtension {
     
     public static final String NAME = "Hashicorp Vault Authentication";
     
-    @Configuration
-    private HashicorpVaultSettings config;
+    @Setting(description = "The token used to access the Hashicorp Vault", key = "edc.vault.hashicorp.token")
```

Comment:
detail, I think this way the setting by default it is required, that's ok in a case where we are using the default token provider, but in case there's another extension registered that overrides it, the setting should not be required.
I would put `required = false` on this and then do a check in the `tokenProvider()` method (that will be called only if no other implementation has been registered)

Reply:
good point 👍🏻